### PR TITLE
guest_os_booting: add new case for multiple boot order attributes

### PR DESCRIPTION
**Pull request by meinaLi**
Automate case: VIRT-297237 - Boot vm with more than one devices with boot order elements

### CODE REVIEW

**Comment by meinaLi** [comment]
```
# avocado run --vt-type libvirt --test-runner=runner --vt-machine-type q35 guest_os_booting.boot_order.boot_with_multiple_boot_order
JOB ID     : 62bb9569a17479b19f6ce185bda09810899b8a5a
JOB LOG    : /var/lib/avocado/job-results/job-2023-11-22T04.49-62bb956/job.log
 (01/12) type_specific.io-github-autotest-libvirt.guest_os_booting.boot_order.boot_with_multiple_boot_order.hd.cdrom.hd_bootable: PASS (42.75 s)
 (02/12) type_specific.io-github-autotest-libvirt.guest_os_booting.boot_order.boot_with_multiple_boot_order.hd.cdrom.cdrom_bootable: PASS (26.96 s)
 (03/12) type_specific.io-github-autotest-libvirt.guest_os_booting.boot_order.boot_with_multiple_boot_order.hd.network.hd_bootable: PASS (43.64 s)
 (04/12) type_specific.io-github-autotest-libvirt.guest_os_booting.boot_order.boot_with_multiple_boot_order.hd.network.network_bootable: PASS (17.63 s)
 (05/12) type_specific.io-github-autotest-libvirt.guest_os_booting.boot_order.boot_with_multiple_boot_order.cdrom.hd.hd_bootable: PASS (43.58 s)
......
 (11/12) type_specific.io-github-autotest-libvirt.guest_os_booting.boot_order.boot_with_multiple_boot_order.network.cdrom.cdrom_bootable: PASS (276.01 s)
 (12/12) type_specific.io-github-autotest-libvirt.guest_os_booting.boot_order.boot_with_multiple_boot_order.network.cdrom.network_bootable: PASS (16.88 s)
RESULTS    : PASS 12 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB HTML   : /var/lib/avocado/job-results/job-2023-11-22T04.49-62bb956/results.html
JOB TIME   : 1115.21 s
```

File: libvirt/tests/src/guest_os_booting/boot_order/boot_with_multiple_boot_order.py [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+    cdrom_dict.update(cdrom_order)
+    cdrom_xml = libvirt_vmxml.create_vm_device_by_type("disk", cdrom_dict)
+    vmxml.add_device(cdrom_xml)
+    vmxml.sync()
+    # Update boot order attributes
+    if "hd" in (first_dev, second_dev):
+        libvirt_vmxml.modify_vm_device(vmxml, 'disk', disk_order)
+    if "network" in (first_dev, second_dev):
+        libvirt_vmxml.modify_vm_device(vmxml, 'interface', network_order)
+    return vmxml, disk_image, cdrom_path
```
[the remark this answers is not in the document]
Comment (meinaLi) [reply libvirt/tests/src/guest_os_booting/boot_order/boot_with_multiple_boot_order.py]: Use a list to record the file paths which need to be deleted at the last.

**Review by chunfuwen — APPROVED** [review]
lgtm

**Comment by meinaLi** [comment]
@Yingshun @dzhengfy @nanli1 Please help to review this PR if any of you have time. This is the last one modular case of guest os booting. Thanks.

**Review by nanli1 — APPROVED** [review]

**Pull request merged**